### PR TITLE
Chore/needed router changes

### PR DESCRIPTION
**Pull request by saledjenic**
Router-related changes that are needed for the release.

Cherry-picked commits for the `status-go` release branch.
- https://github.com/status-im/status-go/pull/5606
- https://github.com/status-im/status-go/pull/5618
- https://github.com/status-im/status-go/pull/5615

### CODE REVIEW

**Comment by J-Son89** [comment]
@saledjenic - could you please give a clearer description of what the problems you have addressed are here and why they were an issue. Additionally it would help if you could link the respective issues. 👌

**Review by jrainville — APPROVED** [review]

File: protocol/communities_messenger_shared_member_address_test.go [hunk 1]
```diff
@@ -819,6 +820,8 @@ func (s *MessengerCommunitiesSharedMemberAddressSuite) TestTokenMasterReceivesAc
 }
 
 func (s *MessengerCommunitiesSharedMemberAddressSuite) TestMemberReceivesPendingRequestToJoinAfterAfterGettingTokenMasterRole() {
+	s.T().Skip("flaky test")
```
Comment (J-Son89) [inline protocol/communities_messenger_shared_member_address_test.go:823]: Should we not just remove flaky tests!?

**Comment by saledjenic** [comment]
> @saledjenic - could you please give a clearer description of what the problems you have addressed are here and why they were an issue. 

@J-Son89 I update it with `Cherry-picked commits for the status-go release branch.`

> Additionally it would help if you could link the respective issues

Related issues were fixed and those commits are already part of the statusgo `develop` branch, I just cherry-picked them here.

**Comment by J-Son89** [comment]
> > @saledjenic - could you please give a clearer description of what the problems you have addressed are here and why they were an issue. 
> 
> 
> 
> @J-Son89 I update it with `Cherry-picked commits for the status-go release branch.`
> 
> 
> 
> > Additionally it would help if you could link the respective issues
> 
> 
> 
> Related issues were fixed and those commits are already part of the statusgo `develop` branch, I just cherry-picked them here.

Thanks for explaining 👌

**Review by J-Son89 — APPROVED** [review]

File: protocol/communities_messenger_shared_member_address_test.go (hunk 1 above)
Comment (saledjenic) [reply protocol/communities_messenger_shared_member_address_test.go:823]: @J-Son89 no, I cannot do that here, cause that commit is one of the commits merged to `develop` before the commit I wanted to cherry-pick here, so I had to include that one as well.

I haven't modified any of the commits you see in the commits tab.

That's why the base branch is `release/0.182.x`, but not `develop`.

**Pull request merged**